src/GraphQL.MicrosoftDI.Tests/FieldBuilderExtensionTests.cs

```diff
@@ -0,0 +1,342 @@
+using System;
+using System.Threading.Tasks;
+using GraphQL.Types;
+using Moq;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.MicrosoftDI.Tests
+{
+    public class FieldBuilderExtensionTests : ScopedContextBase
+    {
+        private readonly ResolveFieldContext _unscopedContext;
+
+        public FieldBuilderExtensionTests()
+        {
+            _scopedServiceProviderMock.Setup(x => x.GetService(It.Is<Type>(x => x == typeof(string)))).Returns("hello").Verifiable();
+            _scopedServiceProviderMock.Setup(x => x.GetService(It.Is<Type>(x => x == typeof(int)))).Returns(2);
+            _scopedServiceProviderMock.Setup(x => x.GetService(It.Is<Type>(x => x == typeof(short)))).Returns((short)3);
+            _scopedServiceProviderMock.Setup(x => x.GetService(It.Is<Type>(x => x == typeof(byte)))).Returns((byte)4);
+            _scopedServiceProviderMock.Setup(x => x.GetService(It.Is<Type>(x => x == typeof(long)))).Returns((long)5);
+            _unscopedContext = new ResolveFieldContext
+            {
+                RequestServices = _scopedServiceProvider
+            };
+        }
+
+        private void VerifyUnscoped()
+        {
+            _scopedServiceProviderMock.Verify();
+        }
+
+        [Fact]
+        public void WithScope1()
+        {
+            var graph = new ObjectGraphType();
+            var field = graph.Field<StringGraphType>()
+                .Resolve()
+                .WithScope()
+                .WithType<string>()
+                .Resolve((context, value) => value)
+                .FieldType;
+            field.Resolver.Resolve(_scopedContext).ShouldBe("hello");
+            VerifyScoped();
+        }
+
+        [Fact]
+        public void WithScope2()
+        {
+            var graph = new ObjectGraphType();
+            var field = graph.Field<StringGraphType>()
+                .Resolve()
+                .WithScope()
+                .WithType<string>()
+                .WithType<int>()
+                .Resolve((context, value, v2) => value + v2)
+                .FieldType;
+            field.Resolver.Resolve(_scopedContext).ShouldBe("hello2");
+            VerifyScoped();
+        }
+
+        [Fact]
+        public void WithScope3()
+        {
+            var graph = new ObjectGraphType();
+            var field = graph.Field<StringGraphType>()
+                .Resolve()
+                .WithScope()
+                .WithType<string>()
+                .WithType<int>()
+                .WithType<short>()
+                .Resolve((context, value, v2, v3) => value + v2 + v3)
+                .FieldType;
+            field.Resolver.Resolve(_scopedContext).ShouldBe("hello23");
+            VerifyScoped();
+        }
+
+        [Fact]
+        public void WithScope4()
+        {
+            var graph = new ObjectGraphType();
+            var field = graph.Field<StringGraphType>()
+                .Resolve()
+                .WithScope()
+                .WithType<string>()
```

Comment:
I think about different name here. What about `WithDependency` or `WithService` ? The type says little to the person who sees this code for the first time. Moreover, in most cases, the type will be the interface.

Reply:
I like either option. `WithService` seems to make the most sense though. ok?

If you have a better suggestion for `WithScope`, lmk.  Perhaps `WithServiceScope` ?  Or maybe leave it?

Reply:
I'd suggest reading the documentation that I wrote and see if `WithScope` feels confusing or not.  Just seemed that I was saying "scope" a lot and not exactly saying what was scoped.  (A scoped service, versus creating a service provider scope for dependency injection)

Even `WithServiceScope` doesn't feel right.  `WithScopedProvider`...?????

Reply:
`WithScope` is ok.

Reply:
Or `InScope`.